http.go

```diff
@@ -203,6 +207,313 @@ func (k *Kasse) GetLogout(res http.ResponseWriter, req *http.Request) {
+func (k *Kasse) GetAddCard(res http.ResponseWriter, req *http.Request) {
+	session, err := k.sessions.Get(req, "nnev-kasse")
+	if err != nil {
+		http.Redirect(res, req, "/login.html", 302)
+		return
+	}
+	ui, ok := session.Values["user"]
+	if !ok {
+		http.Redirect(res, req, "/login.html", 302)
+		return
+	}
+
+	user := ui.(User)
+	data := struct {
+		User        *User
+		Description string
+		Message     string
+	}{
+		User:        &user,
+		Description: "",
+		Message:     "",
+	}
+
+	if err := ExecuteTemplate(res, TemplateInput{Title: "ccchd Kasse", Body: "add_card.html", Data: &data}); err != nil {
+		k.log.Println("Could not render template:", err)
+		http.Error(res, "Internal error", 500)
+		return
+	}
+}
+
+// AddCardEvent returns a json containing the next swiped card UID. The UID is obtained using a channel which is written by the HandleCard method
+func (k *Kasse) AddCardEvent(res http.ResponseWriter, req *http.Request) {
+	session, err := k.sessions.Get(req, "nnev-kasse")
+	if err != nil {
+		http.Redirect(res, req, "/login.html", 302)
+		return
+	}
+	_, ok := session.Values["user"]
+	if !ok {
+		http.Redirect(res, req, "/login.html", 302)
+		return
+	}
+
+	res.Header().Set("Content-Type", "text/event-stream; charset=utf-8")
+	res.WriteHeader(http.StatusOK)
+
+	// Only one go routine can listen on the next card swipe. Tell the client, when it obtains the lock
+	k.registration.Lock()
+	defer k.registration.Unlock()
+	if _, err := res.Write([]byte("event: lock\ndata: lock\n\n")); err != nil {
+		log.Println("Could not write: ", err)
+	}
+	if f, ok := res.(http.Flusher); ok {
+		f.Flush()
+	}
+
+	log.Println("Waiting for Card")
+
+	// Read from the channel for one minute. If the timeout is exceeded and the registration window is still open on the client, the browser reconnects anyway
+	var uid []byte
+	ctx, cancel := context.WithTimeout(req.Context(), 1*time.Minute)
+	defer cancel()
+	select {
+	case uid = <-k.card:
+	case <-ctx.Done():
+		http.Error(res, ctx.Err().Error(), http.StatusRequestTimeout)
+		return
+	}
+
+	// Send card UID in hexadecimal form to client
+	uidString := fmt.Sprintf("%x", uid)
+	log.Println("Card UID obtained! Card uid is", uidString)
+
+	if _, err := res.Write([]byte(fmt.Sprintf("event: card\ndata: %s\n\n", uidString))); err != nil {
+		log.Println("Could not write: ", err)
+	}
+
+	if f, ok := res.(http.Flusher); ok {
+		f.Flush()
+	}
+}
+
+// PostAddCard creates a new Card for the POSTing user
+func (k *Kasse) PostAddCard(res http.ResponseWriter, req *http.Request) {
+	session, err := k.sessions.Get(req, "nnev-kasse")
+	if err != nil {
+		http.Redirect(res, req, "/login.html", 302)
+		return
+	}
+	ui, ok := session.Values["user"]
+	if !ok {
+		http.Redirect(res, req, "/login.html", 302)
+		return
+	}
+
+	user := ui.(User)
+
+	err = req.ParseForm()
+	if err != nil {
+		http.Error(res, "Internal error", http.StatusBadRequest)
+	}
+	description := req.Form.Get("description")
```

Comment:
why not req.FormValue()?

Reply:
Because this is a GET request. The APIs deffer here